ruma-federation-api/src/membership/get_leave_event/v1.rs

```diff
@@ -0,0 +1,79 @@
+//! [GET /_matrix/federation/v1/make_leave/{roomId}/{userId}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-make-leave-roomid-userid)
+
+use std::time::SystemTime;
+
+use ruma_api::ruma_api;
+use ruma_events::{membership::MembershipEventContent, EventType};
+use ruma_identifiers::{RoomId, RoomVersionId, ServerNameBox, UserId};
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "Asks the receiving server to return information that the sending server will need to prepare a leave event to get out of the room.",
+        name: "get_leave_event",
+        method: GET,
+        path: "/_matrix/federation/v1/make_leave/:room_id/:user_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[non_exhaustive]
+    request: {
+        /// The room ID that is about to be left.
+        #[ruma_api(path)]
+        pub room_id: &'a RoomId,
+
+        /// The user ID the leave event will be for.
+        #[ruma_api(path)]
+        pub user_id: &'a UserId,
+    }
+
+    #[non_exhaustive]
+    response: {
+        /// The version of the room where the server is trying to leave. If not provided, the room version is assumed to be either "1" or "2".
+        pub room_version: Option<RoomVersionId>,
+
+        /// An unsigned template event. Note that events have a different format depending on the room version - check the room version specification for precise event formats.
+        pub event: EventTemplate,
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EventTemplate {
+    /// The user ID of the leaving member.
+    pub sender: UserId,
+
+    /// The name of the resident homeserver.
+    pub origin: ServerNameBox,
+
+    /// A timestamp added by the resident homeserver.
+    #[serde(with = "ruma_serde::time::ms_since_unix_epoch")]
+    pub origin_server_ts: SystemTime,
+
+    /// The value `m.room.member`.
+    pub r#type: EventType,
```

Comment:
We usually use `#[serde(rename)]` and rename the field instead of using `r#`. I think this should be `event_type` as in other places.